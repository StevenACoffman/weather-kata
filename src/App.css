.App {
  text-align: center;
}

.App-header {
  background-color: #4c8dff;
  height: 60px;
  padding: 20px;
  color: white;
}
