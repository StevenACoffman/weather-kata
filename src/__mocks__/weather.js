const mockedResponse = {
  "coord": {"lon": -83.61, "lat": 42.24},
  "weather": [{"id": 600, "main": "Snow", "description": "light snow", "icon": "13d"}, {
    "id": 701,
    "main": "Mist",
    "description": "mist",
    "icon": "50d"
  }],
  "base": "stations",
  "main": {"temp": 22.64, "pressure": 1023, "humidity": 85, "temp_min": 21.2, "temp_max": 24.8},
  "visibility": 2414,
  "wind": {"speed": 12.75, "deg": 40, "gust": 9.8},
  "clouds": {"all": 90},
  "dt": 1489443300,
  "sys": {"type": 1, "id": 1382, "message": 0.0071, "country": "US", "sunrise": 1489405660, "sunset": 1489448402},
  "id": 5015688,
  "name": "Ypsilanti",
  "cod": 200
};

export const makeCallAsync = function () {
  return new Promise(resolve => {
    resolve(mockedResponse);
  });
};