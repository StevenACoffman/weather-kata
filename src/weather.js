import request from 'request';

const apiKey = '3a3d115c9f51da324ecfb8a1ad1b7017';
const zip = '48197';

export const makeCallAsync = function() {
  let url = `http://api.openweathermap.org/data/2.5/weather?zip=${zip},us&units=imperial&appid=${apiKey}`;
  return new Promise(resolve => {
    request(url, function (error, response, body) {
      resolve( JSON.parse(body) );
    });
  });
};