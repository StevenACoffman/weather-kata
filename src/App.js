import React, {Component} from 'react';
import './App.css';


class App extends Component {
  constructor(props) {
    super(props);
    this.getWeather = this.getWeather.bind(this);
    this.handleChange = this.handleChange.bind(this);
    this.weather = this.props.weather;
    this.state = {
      weatherBody: '',
    };
  }

  getWeather() {
    this.weather.makeCallAsync(this.state.input)
        .then((response) => {
          this.setState({weatherBody: response});
        });
  }

  handleChange(e) {
    this.setState({input: e.target.value});
  }

  render() {
    return (
        <div className="App">
          <div className="App-header">
            <h2>Weather Kata</h2>
          </div>
          <input placeholder="zip code" onChange={ this.handleChange }/>
          <button onClick={this.getWeather}>
            Get Weather
          </button>
          <p id="city-name">{ this.state.weatherBody.name }</p>
        </div>
    );
  }
}

export default App;
