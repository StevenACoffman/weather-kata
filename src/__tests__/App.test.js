import React from 'react';
import ReactDOM from 'react-dom';
import App from '../App';
import renderer from 'react-test-renderer';


import * as mockedWeather from '../__mocks__/weather';




it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<App />, div);
});

it('calls the mocked api method', () => {
  mockedWeather.makeCallAsync(48197)
    .then(response => {
      expect(response.name).toEqual('Ypsilanti');
    });
});

it('can inject the mock into the App component', () => {
  const div = document.createElement('div');
  const component = renderer.create(<App weather={mockedWeather}/>, div);
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot(); //first snapshot for this test

  tree.children[2].props.onClick(); //manually trigger the button, which is nested
  tree = component.toJSON();
  expect(tree).toMatchSnapshot(); //second snapshot for this test
});

it('uses the other mock syntax', () => {
  const makeCallAsync = jest.fn();
  makeCallAsync.mockReturnValue(new Promise((resolve) => resolve({name: 'South Lyon'})));

  const otherWeather = { makeCallAsync };

  const div = document.createElement('div');
  const component = renderer.create(<App weather={otherWeather}/>, div);
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot(); //first snapshot for this test

  tree.children[2].props.onClick(); //manually trigger the button, which is nested
  tree = component.toJSON();
  expect(tree).toMatchSnapshot(); //second snapshot for this test

});

