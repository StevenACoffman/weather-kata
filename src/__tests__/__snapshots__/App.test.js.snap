exports[`test can inject the mock into the App component 1`] = `
<div
  className="App">
  <div
    className="App-header">
    <h2>
      Weather Kata
    </h2>
  </div>
  <input
    onChange={[Function]}
    placeholder="zip code" />
  <button
    onClick={[Function]}>
    Get Weather
  </button>
  <p />
</div>
`;

exports[`test can inject the mock into the App component 2`] = `
<div
  className="App">
  <div
    className="App-header">
    <h2>
      Weather Kata
    </h2>
  </div>
  <input
    onChange={[Function]}
    placeholder="zip code" />
  <button
    onClick={[Function]}>
    Get Weather
  </button>
  <p />
</div>
`;

exports[`test uses the other mock syntax 1`] = `
<div
  className="App">
  <div
    className="App-header">
    <h2>
      Weather Kata
    </h2>
  </div>
  <input
    onChange={[Function]}
    placeholder="zip code" />
  <button
    onClick={[Function]}>
    Get Weather
  </button>
  <p />
</div>
`;

exports[`test uses the other mock syntax 2`] = `
<div
  className="App">
  <div
    className="App-header">
    <h2>
      Weather Kata
    </h2>
  </div>
  <input
    onChange={[Function]}
    placeholder="zip code" />
  <button
    onClick={[Function]}>
    Get Weather
  </button>
  <p />
</div>
`;
